### ISSUE TITLE
support for iOS 13 devices

### ISSUE THREAD

**Issue by shivamsharma-1996** [socketio/socket.io-client-swift#1237]
iOS Apps build upon Xcode 11 for iOS 13 support need to use swift 5.1 version. As Socket.IO is latest upgraded to swift 5.0, so not compatible to support them.

**Comment by shivamsharma-1996** [comment]
@nuclearace It means you are just expecting .. You dnt have tested this thing?

**please check following issues and get back to me please.**
https://forums.developer.apple.com/thread/117635
https://github.com/AudioKit/AudioKit/issues/1860

**Comment by nuclearace** [comment]
I did a build for ios 13, yes. That is talking about _already_ compiled modules, which won't work. You need to rebuild the framework with the new version of the compiler.

**Comment by shivamsharma-1996** [comment]
@nuclearace Actually socket.IO framework is compiled with swift  5.0 compiler and I'm trying to it compile with **swift 5.1 compiler** under **Xcode 11**.

Reason behind to migrate my project on swift 5.1**(which is only supported by xcode11)**
is to add support for iOS 13 devices also.
**_iOS 13 devices supports the apps written in swift 5.1_**

**Comment by shivamsharma-1996** [comment]
> 
> 
> > It means we need socket.io to compile with swift 5.1 compiler to be able to give support for iOS 13 devices?
> 
> Yes. I was able to do a `pod update` on my test project and build for iOS 13 without a problem.

- Is your test project using socket.io framework?
- you tested this on xcode 11?

**Comment by nuclearace** [comment]
Yes,. my test project is using the framework (installed through CocoaPods) and using Xcode 11.1. 
<img width="1128" alt="Screen Shot 2019-10-17 at 8 15 44 AM" src="https://user-images.githubusercontent.com/1211049/67007912-6d4d3000-f0b6-11e9-9f0c-3bfdd43e21ba.png">

**Comment by shivamsharma-1996** [comment]
Yes you're right. You've used cocoapods.

Actually I've created a framework for VoIP calling in which I'm using **Socket.io & Starscream frameworks** which are using without any **Package manager(like cocoapods) .** 
Instead I used socket.io using following command :
**$ git submodule add https://github.com/socketio/socket.io-client-swift.git**

 Now need to release support for iOS 13 devices. but facing problem with existing socket.io which is used w/o using package manager. 

So can you tell me how it can be resolve?

**Comment by shivamsharma-1996** [comment]
> 
> 
> Yes,. my test project is using the framework (installed through CocoaPods) and using Xcode 11.1.
> <img alt="Screen Shot 2019-10-17 at 8 15 44 AM" width="1128" src="https://user-images.githubusercontent.com/1211049/67007912-6d4d3000-f0b6-11e9-9f0c-3bfdd43e21ba.png">

Can you make sure you don't have used any **toolchain inside xcode 11 to use swift 5.0** 
Also Can you send me the screenshot of swift compiler from target's build settings.

**Comment by nuclearace** [comment]
I created a new project this time, still same result. 
<img width="1401" alt="Screen Shot 2019-10-17 at 9 15 41 AM" src="https://user-images.githubusercontent.com/1211049/67012132-d33db580-f0be-11e9-8d36-bbd0f204b12a.png">
